oolab/src/main/java/agh/ics/oop/MapBoundary.java

```diff
@@ -0,0 +1,30 @@
+package agh.ics.oop;
+import java.util.*;
+
+public class MapBoundary implements IPositionChangeObserver{
+    public List<AbstractWorldMapElement> xEl = new ArrayList<>();
+    public List<AbstractWorldMapElement> yEl = new ArrayList<>();
```

Comment:
same komparatory są ok, ale problem polega na tym, że musimy to wszystko sortować w liście na życzenie więc wydajnościowo zysk jest żaden, a wręcz jest gorzej (bo za każdym razem przy wstawianiu musimy przesortować całą listę N elementów, czyli N*log(N) sprawdzeń. 
Zamiast tego można tu trzymać drzewiastą strukturę, czyli `TreeSet` - wtedy komparator zapinamy tam na stałe i każde wstawienie kosztuje tylko log(N). Nie trzeba też pamiętać o wywolywaniu sortowania, bo drzewo po prostu trzyma pozycje posortowane.